### PR TITLE
BigInt shouldn't be considered experimental

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
I know that in #25886 we discussed that we should gate the functionality behind a flag, but we have a degree of confidence that the BigInt proposal is relatively stable.

- Chrome ships it in stable releases.
- Node 11 ships with it now too

Additionally, we haven't hidden features like object rest/spread behind a flag, even when that feature had no browser implementations, so it's not clear why we need it here.

### CODE REVIEW

**Comment by calebsander** [comment]
I would agree that `BigInt` is not really "experimental"—V8 acts as a well-defined reference implementation. I would add that `BigInt` literals and libs were available in Node 10 (which has been out for 6 months), not just Node 11. Forcing the user to target `esnext` seems like enough of a barrier to check that they really want to use features without universal support. Thanks for fixing my spelling mistake too :smile:

**Comment by sheetalkamat** [comment]
ok. You need to update baselines and also remove experimentalBigInt from bunch of test cases

File: tests/cases/compiler/warnExperimentalBigIntLiteral.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
+// @target: es3
 
 const normalNumber = 123; // should not error
 let bigintType: bigint; // should not error
 let bigintLiteralType: 123n; // should not error when used as type
 let bigintNegativeLiteralType: -123n; // should not error when used as type
-const bigintNumber = 123n * 0b1111n + 0o444n * 0x7fn; // each literal should error
\ No newline at end of file
+const bigintNumber = 123n * 0b1111n + 0o444n * 0x7fn; // each literal should error
+
```
Comment (calebsander) [inline tests/cases/compiler/warnExperimentalBigIntLiteral.ts]: I think you added 2 newlines here, but probably only meant to add 1

File: tests/baselines/reference/warnExperimentalBigIntLiteral.js [hunk 2]
```diff
@@ -3,11 +3,13 @@ const normalNumber = 123; // should not error
...
-const normalNumber = 123; // should not error
-let bigintType; // should not error
-let bigintLiteralType; // should not error when used as type
-let bigintNegativeLiteralType; // should not error when used as type
-const bigintNumber = 123n * 15n + 292n * 0x7fn; // each literal should error
+var normalNumber = 123; // should not error
+var bigintType; // should not error
+var bigintLiteralType; // should not error when used as type
+var bigintNegativeLiteralType; // should not error when used as type
+var bigintNumber = 123n * 15n + 292n * 0x7fn; // each literal should error
```
Comment (calebsander) [inline tests/baselines/reference/warnExperimentalBigIntLiteral.js]: Shouldn't these `warnExperimentalBigIntLiteral.*` baselines be removed?

Comment (calebsander) [reply tests/baselines/reference/warnExperimentalBigIntLiteral.js]: You removed `warnExperimentalBigIntLiteral.errors.txt` but not the `.js`, the `.symbols`, and the `.types` files

**Pull request merged**